compiler/src/dotty/tools/dotc/transform/RepeatableAnnotations.scala

```diff
@@ -0,0 +1,49 @@
+package dotty.tools.dotc
+package transform
+
+import core._
+import ast.tpd._
+import Contexts._
+import MegaPhase._
+import Annotations._
+import Symbols.defn
+import Constants._
+import Types._
+import Decorators._
+
+class RepeatableAnnotations extends MiniPhase:
+  override def phaseName = "repeatableAnnotations"
+
+  override def transformTypeDef(tree: TypeDef)(using Context): Tree = transformDef(tree)
+  override def transformValDef(tree: ValDef)(using Context): Tree = transformDef(tree)
+  override def transformDefDef(tree: DefDef)(using Context): Tree = transformDef(tree)
+
+  private def transformDef(tree: DefTree)(using Context) =
+    val annotations = tree.symbol.annotations
+    if (!annotations.isEmpty) then
+      tree.symbol.annotations = aggregateAnnotations(tree.symbol.annotations)
```

Comment:
Usually, we use `SymTransformer` to change symbols. I think here it's fine to do so as the change is not supposed to be visible in a separately compiled module.